src/server.js

```diff
@@ -0,0 +1,40 @@
+import {WebSocketServer} from 'ws';
+import { v4 as uuid } from 'uuid';
+import ZmqConnection from "./zmq.js";
+import {decodeRawTransaction} from "./helper.js";
+import MessageManager from "./messageManager.js";
+
+export default class Server {
+    constructor() {
+        this.connections = []
+        this.transactions = []
+        this.zmq = new ZmqConnection()
+        this.zmq.on('rawtx', (data) => {
+            this.transactions.push(data);
+        });
+        this.messageManager = new MessageManager()
+    }
+
+    async start() {
+        const socket =  new WebSocketServer({ port: 8000 }); //TODO move magic numbers to env || conf
+
+        socket.on('connection', (socket) => {
+            socket.id = uuid()
+            this.connections.push(socket)
+            socket.send('Welcome to the Dash WebSocket server!');
+        });
+
+        console.log('WebSocket server is running on ws://localhost:8000');
+        setInterval(() => this.handleRawTransactions(), 100)
+    }
+
+    async handleRawTransactions() {
+        if (this.transactions.length) {
+            const decodedTransaction = await decodeRawTransaction(this.transactions.pop())
```

Comment:
We can decode a transaction locally with an dashcore-lib, it will save us reaaaly a lot of time. Making requests is expensive

Reply:
Making async calls are also tricky with the streams, because streams are always synchronous